src/view/components/app.module.ts

```diff
@@ -0,0 +1,197 @@
+import {RuleEngineComponent} from './rule-engine/rule-engine';
+import {RuleEngineContainer} from './rule-engine/rule-engine.container';
+import {ServersideCondition} from './rule-engine/condition-types/serverside-condition/serverside-condition';
+import {VisitorsLocationComponent} from './rule-engine/custom-types/visitors-location/visitors-location.component';
+import {VisitorsLocationContainer} from './rule-engine/custom-types/visitors-location/visitors-location.container';
+
+const RULES_ENGINE_COMPONENTS = [
+    AddToBundleDialogComponent,
+    AddToBundleDialogContainer,
+    AreaPickerDialogComponent,
+    ConditionComponent,
+    ConditionGroupComponent,
+    Dropdown,
+    InputDate,
+    InputOption,
+    InputText,
+    InputToggle,
+    ModalDialogComponent,
+    PushPublishDialogComponent,
+    PushPublishDialogContainer,
+    RestDropdown,
+    RuleActionComponent,
+    RuleComponent,
+    RuleEngineComponent,
+    RuleEngineContainer,
+    ServersideCondition,
+    VisitorsLocationComponent,
+    VisitorsLocationContainer,
+];
+
+// CUSTOM PIPES
+import {CapitalizePipe} from '../../api/pipes/capitalize-pipe';
+
+// CUSTOM DIRECTIVES
+import {MessageKeyDirective} from '../directives/message-keys';
+
+
+const COMPONENTS = [
+    Accordion,
+    AccordionGroup,
+    AppComponent,
+    CustomTimeComponent,
+    DotOption,
+    DotSelect,
+    DropdownComponent,
+    ForgotPasswordComponent,
+    ForgotPasswordContainer,
+    GlobalSearch,
+    IframeLegacyComponent,
+    LoginAsComponent,
+    LoginComponent,
+    LoginContainer,
+    LoginPageComponent,
+    MainComponent,
+    MainNavigation,
+    MyAccountComponent,
+    NotificationsItem,
+    NotificationsList,
+    PatternLibrary,
+    ResetPasswordComponent,
+    ResetPasswordContainer,
+    SiteSelectorComponent,
+    ToolbarAddContenletBodyComponent,
+    ToolbarAddContenletComponent,
+    ToolbarNotifications,
+    ToolbarUserComponent,
+];
+
+const PIPES = [
+    CapitalizePipe
+];
+
+const DIRECTIVES = [
+    MessageKeyDirective
+];
+
+const RULES_ENGINE_SERVICES = [
+    ActionService,
+    ConditionGroupService,
+    ConditionService,
+    GoogleMapService,
+    I18nService,
+    RuleService,
+];
+
+@NgModule({
+    bootstrap: [AppComponent],
+    declarations: [
+        ...COMPONENTS,
+        ...PIPES,
+        ...DIRECTIVES,
+        ...RULES_ENGINE_COMPONENTS,
+    ],
+    imports: [
+        BrowserModule,
+        FormsModule,
+        HttpModule,
+        MaterialModule.forRoot(),
+        ReactiveFormsModule,
+        routing,
+    ],
+    providers: [
+        ApiRoot,
+        RoutingAuthService,
+        BundleService,
+        CoreWebService,
+        DotcmsConfig,
+        DotcmsEventsService,
+        FormatDateService,
+        LoginService,
+        MessageService,
+        NotificationsService,
+        RoutingService,
+        SiteService,
+        UserModel,
+        ...RULES_ENGINE_SERVICES
+    ]
+})
+export class AppModule {
+
+}
```

Comment:
I don't like to much this file, too much imports :D, I think that maybe we could create a new module for rules engine and I don't konw if is a good idea cerate a new modulo too for  login, forgot password, etc